cooperator/models/subscription_request.py

```diff
@@ -843,6 +849,15 @@ def validate_subscription_request(self):
         if self.ordered_parts <= 0:
             raise UserError(_("Number of share must be greater than 0."))
 
+        if self.already_cooperator and self.type != "increase":
+            raise UserError(
+                _(
+                    "Partner %s is already a cooperator, subscription type"
+                    " must be 'Increase number of shares'."
+                )
+                % self.partner_id.name
+            )
+
```

Comment:
>I think this should be considered, I am not sure how the case "existing contact is set" and type of subscription is "new cooperator" is handled, and whether we should prevent it directly from the interface. @huguesdk what do you think ?

nevermind I see it's handled here. It could be nice if it was handled directly in the form, but to me it's OK like this